tmitchell/exercise.py

```diff
@@ -0,0 +1,281 @@
+            new_name = input("")
+            dupe = self.check_duplicate(new_name)
+            if (dupe == 0) & (new_name != self.people[index].name):
+                print("An entry with that name already exists,")
+                print("please try again with a different name")
+            else:
+                self.people[index].edit(new_name)
+
+    def identify_by_name(self, name):
+        """Identify entry then return entry's position in book."""
+        count = 0
+        for i in self.people:
+            if name == i.name:
+                return count
+            count += 1
+        print("No entry found with that name")
+        return "null"
+
+    def check_duplicate(self, name):
+        """Check for entrys with the provided name, returns 0 if already exists or 1 otherwise."""
+        for i in self.people:
+            if name == i.name:
+                return 0
+        return 1
+
+    def order_entries(self):
+        """Sort phonebook alphabetically(ish)."""
+        sortlist = []
+        for i in self.people:
+            if not sortlist:
+                sortlist.append(i)
+            else:
+                counter = len(sortlist)-1
+                while counter >= 0:
+                    if i.name > sortlist[counter].name:
+                        sortlist.insert(counter+1, i)
+                        counter = -1
+                    elif counter == 0:
+                        sortlist.insert(0, i)
+                        counter = -1
+                    else:
+                        counter -= 1
+        self.people = sortlist
+
+    def auto_pop(self):
+        """Fill in phonebook with dummy entries, to be used for testing or demonstration."""
+        print("autopopulating phone book")
+        self.people.append(Person("john", "01245348070", "07800571191"))
+        self.people.append(Person("jonas", "01245348071", "07800571192"))
+        self.people.append(Person("jim", "01245348072", "07800571193"))
+        self.people.append(Person("jerome", "01245348073", "07800571194"))
+        self.people.append(Person("jeremey", "01245348074", "07800571195"))
+        self.people.append(Person("jil", "01245348075", "07800571196"))
+        self.people.append(Person("jeff", "01245348076", "07800571197"))
+        self.people.append(Person("jerry", "01245348077", "07800571198"))
+        self.people.append(Person("josh", "01245348078", "07800571199"))
+        self.people.append(Person("joe", "01245348079", "07800571200"))
+        self.people.append(Person("julia", "01245348080", "07800571201"))
+
+    def save_file(self):
+        """Save current phonebook entries to a file with name input by user."""
+        print("Please enter name of fonbuk to be saved")
+        filename = input("")
+        if filename == "":
+            filename = "DEFAULTSAVE"
+            print("Blank filename has been interpreted as DEFAULTSAVE")
+        file = open(("savefiles/%s" % (filename)), "w")
```

Comment:
In general, try to use `with open(...) as file:` as it will auto close when you are done.